lib/node.js

```diff
@@ -1624,6 +1624,18 @@ declare module "https" {
   declare var globalAgent: Agent;
 }
 
+type module$Module = {
+  builtinModules: Array<string>,
+  createRequire(filename: string | URL): typeof require,
+  syncBuiltinESMExports(): void,
+  Module: module$Module,
```

Comment:
This `Module` property is not documented, but I've somewhat-arbitrarily included it over all the other undocumented properties because ... utility?  Not sure, now that I think about it 🙂.  Let me know if I should remove it.

```
~/flow module²• ❱ node
Welcome to Node.js v15.14.0.
Type ".help" for more information.
> require('module')
<ref *1> [Function: Module] {
  builtinModules: [
    '_http_agent',         '_http_client',        '_http_common',
    '_http_incoming',      '_http_outgoing',      '_http_server',
    '_stream_duplex',      '_stream_passthrough', '_stream_readable',
    '_stream_transform',   '_stream_wrap',        '_stream_writable',
    '_tls_common',         '_tls_wrap',           'assert',
    'assert/strict',       'async_hooks',         'buffer',
    'child_process',       'cluster',             'console',
    'constants',           'crypto',              'dgram',
    'diagnostics_channel', 'dns',                 'dns/promises',
    'domain',              'events',              'fs',
    'fs/promises',         'http',                'http2',
    'https',               'inspector',           'module',
    'net',                 'os',                  'path',
    'path/posix',          'path/win32',          'perf_hooks',
    'process',             'punycode',            'querystring',
    'readline',            'repl',                'stream',
    'stream/promises',     'string_decoder',      'sys',
    'timers',              'timers/promises',     'tls',
    'trace_events',        'tty',                 'url',
    'util',                'util/types',          'v8',
    'vm',                  'wasi',                'worker_threads',
    'zlib'
  ],
  _cache: [Object: null prototype] {},
  _pathCache: [Object: null prototype] {},
  _extensions: [Object: null prototype] {
    '.js': [Function (anonymous)],
    '.json': [Function (anonymous)],
    '.node': [Function (anonymous)]
  },
  globalPaths: [
    '/Users/isker/.node_modules',
    '/Users/isker/.node_libraries',
    '/usr/local/Cellar/node/15.14.0/lib/node'
  ],
  _debug: [Function: deprecated],
  _findPath: [Function (anonymous)],
  _nodeModulePaths: [Function (anonymous)],
  _resolveLookupPaths: [Function (anonymous)],
  _load: [Function (anonymous)],
  _resolveFilename: [Function (anonymous)],
  createRequireFromPath: [Function: deprecated],
  createRequire: [Function: createRequire],
  _initPaths: [Function (anonymous)],
  _preloadModules: [Function (anonymous)],
  syncBuiltinESMExports: [Function: syncBuiltinESMExports],
  Module: [Circular *1],
  runMain: [Function: executeUserEntryPoint],
  findSourceMap: [Function: findSourceMap],
  SourceMap: [class SourceMap]
}

```